src/lsm/compaction.zig

```diff
@@ -511,6 +511,11 @@ pub fn CompactionType(
                 // Assert that we're reading data blocks in key order.
                 const values_in = compaction.values_in[index];
                 if (values_in.len > 0) {
+                    if (constants.verify) {
+                        for (values_in[0 .. values_in.len - 1], values_in[1..]) |*value, *value_next| {
```

Comment:
I was wondering whether this also changes the overall complexity here, like with the binary search, but I think this is fine. We do this once for data block, so this is O(N). 

Though, I am wondering --- why we need `values_in.len > 0` conditional? I would assume that data blocks are always not-empty

Reply:
I switched the branch to an assert after the discussion. The {tree, forest} fuzzers appear to still pass, so it seems like it's good.